From ab6a1e31883c05e142ee4d1486c19929a08f122f Mon Sep 17 00:00:00 2001
From: Liu Ying <Ying.liu@freescale.com>
Date: Tue, 11 Sep 2012 18:20:39 +0800
Subject: ENGR00223797-3 ARM:IPUv3 fb:Add late init in pdata

This patch adds late init field support in ipuv3 fb
platform data, so that a platform may choose to not
to initialize framebuffer until the user triggers
set_par(), which may support smooth transition from
bootloader splashimage to system UI.

Signed-off-by: Liu Ying <Ying.Liu@freescale.com>
(cherry picked from commit 3a2547ac167fb2c06d9f63c7a925b9ebdc896d96)
---
 arch/arm/plat-mxc/include/mach/ipu-v3.h |    7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/arch/arm/plat-mxc/include/mach/ipu-v3.h b/arch/arm/plat-mxc/include/mach/ipu-v3.h
index 48daf4d..519e628 100755
--- a/arch/arm/plat-mxc/include/mach/ipu-v3.h
+++ b/arch/arm/plat-mxc/include/mach/ipu-v3.h
@@ -737,6 +737,13 @@ struct ipuv3_fb_platform_data {
 	/* reserved mem */
 	resource_size_t 		res_base[2];
 	resource_size_t 		res_size[2];
+
+	/*
+	 * Late init to avoid display channel being
+	 * re-initialized as we've probably setup the
+	 * channel in bootloader.
+	 */
+	bool                            late_init;
 };
 
 struct imx_ipuv3_platform_data {
-- 
1.7.9.5

